Source/Scene/Cesium3DTileFeatureTableResources.js

```diff
@@ -0,0 +1,87 @@
+/*global define*/
+define([
+        '../Core/defaultValue',
+        '../Core/defined',
+        '../Core/DeveloperError',
+        '../Renderer/WebGLConstants'
+    ], function(
+        defaultValue,
+        defined,
+        DeveloperError,
+        WebGLConstants) {
+    'use strict';
+
+    /**
+     * @private
+     */
+    function Cesium3DTileFeatureTableResources(featureTableJSON, featureTableBinary) {
+        this.json = featureTableJSON;
+        this.binaryDataView = new DataView(featureTableBinary);
+        this._cachedProperties = {};
+        this.featuresLength = 0;
+    }
+
+    Cesium3DTileFeatureTableResources.prototype.getGlobalProperty = function(semantic, componentType, count) {
+        if (defined(this._cachedProperties[semantic])) {
+            return this._cachedProperties[semantic];
+        } else {
+            var jsonValue = this.json[semantic];
+            if (defined(jsonValue)) {
+                var result;
+                if (defined(jsonValue.byteOffset)) {
+                    //>>includeStart('debug', pragmas.debug);
+                    if (!defined(componentType)) {
+                        throw new DeveloperError('componentType must be defined to read from binary data for semantic: ' + semantic);
+                    }
+                    //>>includeEnd('debug');
+                    var byteOffset = jsonValue.byteOffset;
+                    // This is a reference to the binary
+                    count = defaultValue(count, 1);
+                    if (count > 1) {
+                        result = [];
+                        var componentByteLength = byteLengthForComponentType(componentType);
+                        for (var i = 0; i < count; i++) {
```

Comment:
There is a much much faster way to do this with typed arrays.  For example, see https://github.com/AnalyticalGraphicsInc/cesium/blob/master/Source/Core/ComponentDatatype.js#L227